internal/rpcserver/rpcserver.go

```diff
@@ -3564,6 +3569,7 @@ func handleGetTxOut(_ context.Context, s *Server, cmd interface{}) (interface{},
 		scriptVersion = txOut.Version
 		pkScript = txOut.PkScript
 		isCoinbase = standalone.IsCoinBaseTx(mtx, isTreasuryEnabled)
+		txType = stake.DetermineTxType(mtx, isTreasuryEnabled)
```

Comment:
The tree is already available via `txFromMempool.Tree()`.  I wonder if the `txTree` should be declared above and then set directly here instead?  Then in the other case, it would use the "txTypeToTree" logic.

I think it's fine either way, but since `stake.DetermineTxType` is somewhat of an expensive call, it would certainly be faster to avoid it.

Reply:
Ah, ya, that is a good call.  I'll update it to that.  Also, that will work out well since the "txTypeToTree" in the other case will be temporary and will be removed after the utxo lookup is updated to include the tree (so the entry would already be nil if the tree didn't match).